src/results/state/actions.js

```diff
@@ -0,0 +1,29 @@
+import queryString from 'query-string';
+import urljoin from 'url-join';
+import fetchData from '../../utils/fetchData';
+import apiUrls from '../../utils/apiUrls';
+import createQueryString from '../utils/QueryStringGenerator';
+
+export const FETCH_RESULTS_STARTED = 'FETCH_RESULTS_STARTED';
+export const FETCH_RESULTS_SUCCESS = 'FETCH_RESULTS_SUCCESS';
+
+export const fetchResultsSuccess = data => ({
+  type: FETCH_RESULTS_SUCCESS,
+  data,
+  receivedAt: Date.now(),
+});
+
+export const fetchResultsStarted = () => ({
+  type: FETCH_RESULTS_STARTED,
+});
+
+export const fetchResults = query => (dispatch) => {
+  const uniprotQueryString = createQueryString(query);
+  const queryUrl = queryString.stringify({
+    query: uniprotQueryString,
+  });
+  dispatch(fetchResultsStarted());
+  fetchData(urljoin(apiUrls.advanced_search_terms, '/?', queryUrl))
+    .then(response => dispatch(fetchResultsSuccess(response.data)))
+    .catch(error => throw error);
```

Comment:
Will want to add `@babel/plugin-proposal-throw-expressions` to dev deps & babelrc to support the `throw` statement.

Reply:
We still need to decide on error handling in the application. Reverting to console.error for now